pkg/ring/kv/memberlist/memberlist_client.go

```diff
@@ -524,7 +545,12 @@ outer:
 		if change != nil {
 			m.casSuccesses.Inc()
 			m.notifyWatchers(key)
-			m.broadcastNewValue(key, change, newver, codec)
+
+			if m.casBroadcastsEnabled.Load() {
+				m.broadcastNewValue(key, change, newver, codec)
+			} else {
+				level.Error(util.Logger).Log("msg", "CAS with broadcasts disabled", "key", key)
```

Comment:
- Is this an error? Doesn't look actionable to me. Maybe `Warn` is better?
- What do you think if we change the log message into something like `skipped broadcasting CAS message because broadcasting is disabled` for clarity?

Reply:
> * Is this an error? Doesn't look actionable to me. Maybe `Warn` is better?

It's an error in the code, in a sense that it should never happen. We stop writers before KV. But from user's point of view, it's not something they can do anything with, so I'll make it a warning.

> * What do you think if we change the log message into something like `skipped broadcasting CAS message because broadcasting is disabled` for clarity?

Sounds good

Reply:
What about: `skipped broadcasting CAS update because memberlist KV is shutting down` ?

Reply:
> What about: `skipped broadcasting CAS update because memberlist KV is shutting down` ?

Even better. But then what if we rename `casBroadcastsEnabled` to `shuttingDown`? May be more clear to the code reader.

Reply:
I like `casBroadcastsEnabled` more, because it's specific. `shuttingDown` may imply more than it really does.